docs/rfcs/rfc-040-relay-first-api.md

```diff
@@ -0,0 +1,1282 @@
+    ): MovieActorsConnection!
+    director(where: MovieDirectorConnectionWhere, directed: Boolean = true): MovieDirectorConnection!
+}
+```
+
+Note that this type does **not** contain any operations in itself, it has the same properties as the user-defined entity, with the only difference being the result of the nested `actors` and `director` types, that support the top-level input and return a `*Connection`.
+
+### Filtering
+
+Filtering follows the same structure as projections, following nodes and edges. The operations `AND`, `OR` and `NOT` are available at any level. Comparison operators are present inside each field.
+
+```graphql
+query MatrixMoviesFrom1999 {
+    moviesConnection(
+        where: { edges: { node: { AND: [{ title: { contains: "Matrix" } }, { released: { eq: 1999 } }] } } }
+    ) {
+        edges {
+            node {
+                title
+            }
+        }
+    }
+}
+```
+
+#### Filtering Relationships
+
+For filtering edges, we can use the `fields` property and the array operators if needed:
+
+```graphql
+query PeopleAndMoviesActedAfter2001 {
+    peopleConnection(where: { edges: { node: { movies: { edges: { some: { fields: { year: { gt: 1999 } } } } } } } }) {
+        edges {
+            node {
+                name
+                movies {
+                    edges {
+                        node {
+                            title
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+Both, arrays and relationships with many elements can use the operators `some`, `single`, `none` and `all` to filter.
+
+#### Filter on nested connections
+
+Nested relationships can be filtered in the same fashion as top level connections, in this case only applies to the nested elements:
+
+```graphql
+query PeopleAndMoviesWithNestedFilter {
+    peopleConnection {
+        edges {
+            node {
+                name
+                movies(where: { edges: { node: { title: { equals: "The Matrix" } } } }) {
+                    edges {
+                        node {
+                            title
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+### Sort And Pagination
+
+Pagination, by default, will use `cursor` pagination, according to the Relay spec:
+
+```graphql
+query PaginatedMovies {
+    moviesConnection(first: 10, after: "my-cursor", sort: { edges: { node: { title: DESC } } }) {
+        edges {
+            node {
+                title
+            }
+            cursor
+        }
+        pageInfo {
+            startCursor
+            hasPreviousPage
+            hasNextPage
+            endCursor
+        }
+    }
+}
+```
+
+Following Relay spec, every `*Connection` type has the `pageInfo` property, and each `edge` has a cursor. Note that if limit-based pagination is enabled instead of cursor based, the API will not be Relay compliant.
```

Comment:
Is it possible to see an example of limit-based pagination with the new API? It'd also be good to see how you propose to allow users to configure the pagination type

 Also, this is a bit nit picky but is it not more commonly called offset-based?

Reply:
I'll add an example and description in its own section.

I will not add how the configuration would work, as that would fall into how we plan to do server configuration, which is out of scope for this API design